domain/schema/model.go

```diff
@@ -412,6 +412,7 @@ CREATE TABLE machine (
     machine_id      TEXT NOT NULL,
     net_node_uuid   TEXT NOT NULL,
     life_id         INT NOT NULL,
+    keep_instance   BOOLEAN,
```

Comment:
I'd suggest taking out this `keep_instance` field entirely. Looks like this is only used in the `remove-machine` to keep the cloud instance running, which suggests that it's an operational concern, and not a modelling concern. I don't see why we currently have it in the `state.instanceData`, it's not used for the [`--keep-broken` in the bootstrap](https://github.com/juju/juju/blob/86515ba06675afa6f67e9ec1ac2d03b6481a4c99/cmd/juju/commands/bootstrap.go#L963) either.

Reply:
What do you mean by "operational concern"? If you were thinking about having these as an in memory variable it's not possible because cleanup is asynchronous and we couldn't guarantee in case of a controller reboot, etc. 
Maybe I'm misunderstanding and you want the whole feature to be removed?

Reply:
It is a modelling concern, because Juju records desired state, and agents act to realise it.

When calling `remove-machine` we need to record the desired state, which is what this does. The provisioner and the container broker see this desired state declaration and act accordingly to de-provision the machine/container without removing the instance.